src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionPresenter.cs

```diff
@@ -45,7 +45,7 @@ internal sealed class CompletionPresenter : ForegroundThreadAffinitizedObject, I
 
             var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
 
-            var completionSetFactory = document != null && NeedsDev15CompletionSetFactory(document.Options, document.Project.Language)
+            var completionSetFactory = document != null && NeedsDev15CompletionSetFactory(document.Project.Solution.Options, document.Project.Language)
```

Comment:
Actually I'm going to leave this: looks like we have a big of a silly bug historically where options could change while we're asynchronously doing stuff, and places that requested options that way would be bad. This will let us reestablish snapshot semantics. It doesn't do that yet, but once I had a WorkspaceChangedKind for options changed, we'll fix this up.